tests/src/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -103,14 +126,64 @@ public class CoreclrTestWrapperLib
 
         static bool CollectCrashDump(Process process, string path)
         {
-            using (var crashDump = File.OpenWrite(path))
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                using (var crashDump = File.OpenWrite(path))
+                {
+                    var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
+                    return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+                }
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
             {
-                var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
-                return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+                string coreRoot = Environment.GetEnvironmentVariable("CORE_ROOT");
+                ProcessStartInfo createdumpInfo = new ProcessStartInfo("sudo");
+                createdumpInfo.Arguments = $"{Path.Combine(coreRoot, "createdump")} --name \"{path}\" {process.Id} -h";
+                Process createdump = Process.Start(createdumpInfo);
+                return createdump.WaitForExit(DEFAULT_TIMEOUT) && createdump.ExitCode == 0;
```

Comment:
createdump so far has to exist alongside the runtime, which means that looking for it in `CORE_ROOT` should be safe. It's a contract that's assumed by diagnostic scenarios at large. I guess there could be some error checking to prevent an unhandled exception here with a log on the issue.

Reply:
Since we're running it through sudo, sudo actually errors out gracefully here if createdump isn't found and the code handles the failure.